File: javascript/283/move_zeros.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    // console.log(nums)
    var k = 0;
    for (var i = 0; i < nums.length; i++){
        if (nums[i]){
            if (i != k){
                var temp;
                temp = nums[i];
                nums[i] = nums[k];
                nums[k] = temp;
                // console.log(nums);
                k++;
            }
            else{
                k++;
            }
            
        }
    }

};

File: python/283/move_zeros.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        k = 0
        temp = 0
        for i in range(0, len(nums)):
            if (nums[i]):
                temp = nums[i]
                nums[i] = nums[k]
                nums[k] = temp
                k += 1
